### DISCUSSION TITLE
infinispan hotrod connector on multi subnet

### DISCUSSION THREAD

**Discussion by cyrilfantin** [infinispan/infinispan#12132] [General]
Hello
I try to deploy infinispan (latest version of 14) on AWS ECS and connect a keycloak (23.0.X) with the infinispan cluster.

I succed to deploy an infinispan cluster over 2 differents subnets. It works fine. 
But when i try to connect keycloak to the infinispan cluster the hotrod connector cannot find one of the node of infinispan : 

org.infinispan.client.hotrod.exceptions.HotRodClientException:Request for messageId=6 returned server error (status=0x85): java.lang.IllegalArgumentException: No interface address matching '/172.64.2.72' in MultiHomedServerAddress{port=11222, addresses=[/172.64.2.205/25, /169.254.172.2/22, /127.0.0.1/8]}

I try to deploy infinispan on one subnet and connect keycloak to it(2 nodes on differents subnets) and it works fine.

The problem is when the infinispan cluster is on 2 differents subnets. It uses the netmasks /25 and i want to configure it with /24

Is there a configuration to achieve that ?

if possible i don't want to configure a cross site configuration, it will be hard on AWS ECS...

Thank you

**Comment by tristantarrant** [comment]
You could bind directly to the specific address, instead of binding to `0.0.0.0`

**Comment by cyrilfantin** [comment]
ok, i found a way just after this post...
i set the external-host of hotrod-connector with the url of the service discovery (i'm using DNS_PING as resolver).
It seems working fine.

Thank you
